<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1651</width>
    <height>1019</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>icons/chipdrawer.png</normaloff>icons/chipdrawer.png</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QFrame" name="frame">
        <property name="minimumSize">
         <size>
          <width>400</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>400</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Chip Properties</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_7" columnstretch="1,2,0">
            <item row="1" column="2">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>mm</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>Height</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Margins</string>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>mm</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="spinner_margins">
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="value">
               <number>2</number>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="spinner_height">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="value">
               <number>10</number>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Width</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="spinner_width">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="value">
               <number>20</number>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>mm</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>Layers</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QPushButton" name="btn_add_layer">
                <property name="text">
                 <string>Add</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btn_remove_layer">
                <property name="text">
                 <string>Remove</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QTableView" name="layer_list">
              <property name="alternatingRowColors">
               <bool>true</bool>
              </property>
              <property name="selectionMode">
               <enum>QAbstractItemView::SingleSelection</enum>
              </property>
              <property name="selectionBehavior">
               <enum>QAbstractItemView::SelectRows</enum>
              </property>
              <property name="showGrid">
               <bool>false</bool>
              </property>
              <property name="gridStyle">
               <enum>Qt::NoPen</enum>
              </property>
              <property name="wordWrap">
               <bool>false</bool>
              </property>
              <attribute name="horizontalHeaderMinimumSectionSize">
               <number>30</number>
              </attribute>
              <attribute name="horizontalHeaderDefaultSectionSize">
               <number>70</number>
              </attribute>
              <attribute name="horizontalHeaderHighlightSections">
               <bool>false</bool>
              </attribute>
              <attribute name="horizontalHeaderStretchLastSection">
               <bool>true</bool>
              </attribute>
              <attribute name="verticalHeaderVisible">
               <bool>false</bool>
              </attribute>
              <attribute name="verticalHeaderHighlightSections">
               <bool>false</bool>
              </attribute>
             </widget>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QPushButton" name="btn_layer_up">
                <property name="text">
                 <string>Up</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="btn_layer_down">
                <property name="text">
                 <string>Down</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="layer_props">
           <property name="title">
            <string>Layer Properties</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_8">
            <item row="4" column="0">
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Thickness</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="layer_prop_name"/>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Name</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="layer_prop_material"/>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Material</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QDoubleSpinBox" name="layer_prop_thickness"/>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Visible</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QCheckBox" name="layer_prop_visible">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Substrate</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QCheckBox" name="layer_prop_substrate">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="CDProject" name="drawing_area">
        <property name="mouseTracking">
         <bool>true</bool>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOn</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOn</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="frame_2">
        <property name="minimumSize">
         <size>
          <width>380</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>380</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QGridLayout" name="gridLayout_2">
         <item row="0" column="0">
          <widget class="QToolBox" name="toolBox">
           <property name="currentIndex">
            <number>0</number>
           </property>
           <widget class="QWidget" name="page_3">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>360</width>
              <height>834</height>
             </rect>
            </property>
            <attribute name="label">
             <string>Standard Blocks</string>
            </attribute>
            <layout class="QGridLayout" name="gridLayout_3">
             <item row="0" column="0">
              <widget class="QScrollArea" name="scrollArea">
               <property name="styleSheet">
                <string notr="true">QPushButton{background-color:white;}</string>
               </property>
               <property name="widgetResizable">
                <bool>true</bool>
               </property>
               <widget class="QWidget" name="standard_blocks_layout">
                <property name="geometry">
                 <rect>
                  <x>0</x>
                  <y>0</y>
                  <width>340</width>
                  <height>814</height>
                 </rect>
                </property>
                <layout class="QGridLayout" name="gridLayout_4"/>
               </widget>
              </widget>
             </item>
            </layout>
           </widget>
           <widget class="QWidget" name="page_2">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>0</y>
              <width>360</width>
              <height>834</height>
             </rect>
            </property>
            <attribute name="label">
             <string>Default blocks</string>
            </attribute>
            <layout class="QGridLayout" name="gridLayout_5">
             <item row="0" column="0">
              <widget class="QScrollArea" name="scrollArea_2">
               <property name="styleSheet">
                <string notr="true">QPushButton{background-color:white;}</string>
               </property>
               <property name="widgetResizable">
                <bool>true</bool>
               </property>
               <widget class="QWidget" name="custom_blocks_layout">
                <property name="geometry">
                 <rect>
                  <x>0</x>
                  <y>0</y>
                  <width>340</width>
                  <height>814</height>
                 </rect>
                </property>
                <property name="styleSheet">
                 <string notr="true"/>
                </property>
                <layout class="QGridLayout" name="gridLayout_6"/>
               </widget>
              </widget>
             </item>
            </layout>
           </widget>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1651</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <widget class="QMenu" name="menuOpen_Recent">
     <property name="title">
      <string>Open Recent...</string>
     </property>
     <addaction name="action_None"/>
    </widget>
    <addaction name="action_New"/>
    <addaction name="action_Open"/>
    <addaction name="menuOpen_Recent"/>
    <addaction name="separator"/>
    <addaction name="action_Save"/>
    <addaction name="action_SaveAs"/>
    <addaction name="separator"/>
    <addaction name="action_Export"/>
    <addaction name="action_Print"/>
    <addaction name="separator"/>
    <addaction name="action_Exit"/>
   </widget>
   <widget class="QMenu" name="menu_Edit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="separator"/>
    <addaction name="action_Cut"/>
    <addaction name="action_Copy"/>
    <addaction name="action_Paste"/>
    <addaction name="action_Delete"/>
    <addaction name="action_Select_All"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Edit"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="minimumSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="baseSize">
    <size>
     <width>0</width>
     <height>0</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="iconSize">
    <size>
     <width>32</width>
     <height>32</height>
    </size>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonIconOnly</enum>
   </property>
   <property name="floatable">
    <bool>false</bool>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_New"/>
   <addaction name="action_Open"/>
   <addaction name="action_Save"/>
   <addaction name="separator"/>
   <addaction name="action_Cut"/>
   <addaction name="action_Copy"/>
   <addaction name="action_Paste"/>
   <addaction name="separator"/>
   <addaction name="action_Delete"/>
   <addaction name="action_Select_All"/>
   <addaction name="separator"/>
   <addaction name="action_3D"/>
  </widget>
  <action name="action_New">
   <property name="icon">
    <iconset>
     <normaloff>icons/icon_new.png</normaloff>icons/icon_new.png</iconset>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_Open">
   <property name="icon">
    <iconset>
     <normaloff>icons/icon_open.png</normaloff>icons/icon_open.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="icon">
    <iconset>
     <normaloff>icons/icon_save.png</normaloff>icons/icon_save.png</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_SaveAs">
   <property name="text">
    <string>Save &amp;As</string>
   </property>
   <property name="shortcut">
    <string>F12</string>
   </property>
  </action>
  <action name="action_Export">
   <property name="text">
    <string>&amp;Export</string>
   </property>
  </action>
  <action name="action_Print">
   <property name="text">
    <string>&amp;Print</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="action_Exit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
  </action>
  <action name="action_3D">
   <property name="text">
    <string>3D</string>
   </property>
   <property name="shortcut">
    <string>F3</string>
   </property>
  </action>
  <action name="action_None">
   <property name="text">
    <string>None</string>
   </property>
  </action>
  <action name="action_Cut">
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="action_Copy">
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="action_Paste">
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="action_Delete">
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="action_Select_All">
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>CDProject</class>
   <extends>QGraphicsView</extends>
   <header>cdproject.project</header>
   <slots>
    <slot>signal_width_changed(int)</slot>
    <slot>signal_height_changed(int)</slot>
    <slot>signal_margins_changed(int)</slot>
    <slot>signal_layer_add()</slot>
    <slot>signal_layer_remove()</slot>
    <slot>signal_layer_clicked(QModelIndex)</slot>
    <slot>signal_cut()</slot>
    <slot>signal_copy()</slot>
    <slot>signal_paste()</slot>
    <slot>signal_delete()</slot>
    <slot>signal_select_all()</slot>
    <slot>signal_layer_changed_material(int)</slot>
    <slot>signal_layer_changed_name()</slot>
    <slot>signal_layer_changed_visibility(bool)</slot>
    <slot>signal_layer_changed_substrate(bool)</slot>
    <slot>signal_layer_changed_thickness()</slot>
    <slot>signal_layer_up()</slot>
    <slot>signal_layer_down()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>spinner_height</sender>
   <signal>valueChanged(int)</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_height_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>253</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinner_width</sender>
   <signal>valueChanged(int)</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_width_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>253</x>
     <y>104</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinner_margins</sender>
   <signal>valueChanged(int)</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_margins_changed(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>253</x>
     <y>158</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_add_layer</sender>
   <signal>clicked()</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_layer_add()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>118</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_remove_layer</sender>
   <signal>clicked()</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_layer_remove()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>223</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>layer_list</sender>
   <signal>clicked(QModelIndex)</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_layer_clicked(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>209</x>
     <y>605</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>522</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Exit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>825</x>
     <y>509</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Cut</sender>
   <signal>triggered()</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_cut()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Copy</sender>
   <signal>triggered()</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_copy()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Paste</sender>
   <signal>triggered()</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_paste()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Delete</sender>
   <signal>triggered()</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_delete()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Select_All</sender>
   <signal>triggered()</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_select_all()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>layer_prop_material</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_layer_changed_material(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>886</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>layer_prop_name</sender>
   <signal>editingFinished()</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_layer_changed_name()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>859</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>layer_prop_visible</sender>
   <signal>clicked(bool)</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_layer_changed_visibility(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>910</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>layer_prop_substrate</sender>
   <signal>clicked(bool)</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_layer_changed_substrate(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>932</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>layer_prop_thickness</sender>
   <signal>editingFinished()</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_layer_changed_thickness()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>238</x>
     <y>957</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_layer_up</sender>
   <signal>clicked()</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_layer_up()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>118</x>
     <y>793</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_layer_down</sender>
   <signal>clicked()</signal>
   <receiver>drawing_area</receiver>
   <slot>signal_layer_down()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>793</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>530</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>signal_add_layer()</slot>
  <slot>signal_remove_layer()</slot>
  <slot>signal_layer_clicked(QModelIndex)</slot>
  <slot>signal_show_3d()</slot>
  <slot>signal_edit_layer()</slot>
  <slot>signal_save()</slot>
  <slot>signal_export()</slot>
  <slot>signal_open()</slot>
 </slots>
</ui>
